src/test/java/org/opensearch/flowframework/indices/GlobalContextHandlerTests.java

```diff
@@ -0,0 +1,109 @@
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.workflow.CreateIndexStep;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class GlobalContextHandlerTests extends OpenSearchTestCase {
+    @Mock
+    private Client client;
+    @Mock
+    private CreateIndexStep createIndexStep;
+    @Mock
+    private ThreadPool threadPool;
+    private GlobalContextHandler globalContextHandler;
+    private AdminClient adminClient;
+    private IndicesAdminClient indicesAdminClient;
+    private ThreadContext threadContext;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        MockitoAnnotations.openMocks(this);
+
+        Settings settings = Settings.builder().build();
+        threadContext = new ThreadContext(settings);
+        when(client.threadPool()).thenReturn(threadPool);
+        when(threadPool.getThreadContext()).thenReturn(threadContext);
+
+        globalContextHandler = new GlobalContextHandler(client, createIndexStep);
+        adminClient = mock(AdminClient.class);
+        indicesAdminClient = mock(IndicesAdminClient.class);
+        when(adminClient.indices()).thenReturn(indicesAdminClient);
+        when(client.admin()).thenReturn(adminClient);
+    }
+
+    public void testPutTemplateToGlobalContext() throws IOException {
+        Template template = mock(Template.class);
+        when(template.toXContent(any(XContentBuilder.class), eq(ToXContent.EMPTY_PARAMS))).thenAnswer(invocation -> {
+            XContentBuilder builder = invocation.getArgument(0);
+            return builder;
+        });
+        ActionListener<IndexResponse> listener = mock(ActionListener.class);
+
+        doAnswer(invocation -> {
+            ActionListener<Boolean> callback = invocation.getArgument(1);
+            callback.onResponse(true);
+            return null;
+        }).when(createIndexStep).initIndexIfAbsent(any(), any());
+
+        globalContextHandler.putTemplateToGlobalContext(template, listener);
+
+        ArgumentCaptor<IndexRequest> requestCaptor = ArgumentCaptor.forClass(IndexRequest.class);
+        verify(client).index(requestCaptor.capture(), any());
```

Comment:
Nit/Style: I like to add `times` with verify for specific number of invocations :)
```suggestion
        verify(client, times(1)).index(requestCaptor.capture(), any());
```